build.sbt

```diff
@@ -394,6 +394,8 @@ lazy val spark = (project in file("connectors/spark"))
     ),
     Compile / unmanagedJars += (serverAndClientShaded / assembly).value,
     Test / unmanagedJars += (serverAndClientShaded / assembly).value,
+    // TODO: enable the java assertion after Delta 3.2.1 is released.
+    // Test / javaOptions += "-ea",
```

Comment:
@creechy @alexreid-db can you make sure that this is merged in a separate PR to fix the infra, all projects need to have this set so that we dont silently ignore test failures. 

if there are tests that are currently failing, lets mark them as ignored, and then we will fix them one by one in follow up PRs like this one.

Reply:
Created https://github.com/unitycatalog/unitycatalog/pull/350 to address this